### ISSUE TITLE
RAML regex pattern properties: not formatted in Query Parameters

### ISSUE THREAD

**Issue by stmeissner** [commercetools/commercetools-docs-kit#1009]
![Screen Shot 2021-07-06 at 11 09 05](https://user-images.githubusercontent.com/900598/124574531-d9536700-de4a-11eb-800a-2afeb96ada3b.png)

Expected result is that it's rendered like in a payload representation type:
https://commercetools-docs-kit.vercel.app/api-docs-smoke-test/types/general#regexpropertytesttype

Here's the existing docs kit test case:  https://commercetools-docs-kit.vercel.app/api-docs-smoke-test/endpoints/methods#get-search-resource  (not formatted yet). 

----
second issue in the same lines of code

The implementation has a bug but we oversaw it when initially merging the feature:  The data type of the respective field or parameter is lost,  the text "Any property matching regular expression above" is _replacing_ the  type information ("String" / "Integer" / etc.) instead of being added to it. 

This poses a little visual problem:  How to fit that sentence **and** the Type?   A solution that would not require a design change would be to make the text include the type.  For example:  "Any String property matching regular expression above"  //  "Any String parameter matching the regular expression above".  

Or, a more concise wording:  "Any String property matching this regular expression"  ("above" would be misleading in the query parameter layout anyways because it's next to  it, not below). 

----

plus a third issue: The text has a grammar issue, it's missing a "the":   "Any parameter matching **the** regular expression above"

**Comment by nkuehn** [comment]
@stmeissner  It took a bit but now I did the agreed update and clarification to the issue text (directly done in the description), renamed it and put it on the board. 

We need to take a wording decision, my candidates are
 1. "Any (String|Integer|Float|...) (property|parameter) matching this regular expression"
 2. "Any (property|parameter) matching this regular expression - (String|Integer|Float|...)"
 2. "(String|Integer|Float|...) - Any (property|parameter) matching this regular expression"
 3. "Any (String|Integer|Float|...) (property|parameter) matching the regular expression above"  (plus the corresponding variations)

**Comment by stmeissner** [comment]
Thanks for the update.
I would think that regular expressions are for String types only, so we can restrict it to those?
I suggest: "Any (property/parameter) matching this regular expression."

When looking at the screenshot, I find it difficult to understand what the last parameter would be. Isn't the RegEx rather meant to specify the possible values of the `where` parameter? So, it should not appear in the list of query parameters then, right?

**Comment by nkuehn** [comment]
The regular expression describes the allowed *names* of the properties or parameters, but not the value.   For example, an API could have a pattern property `/number{0-9}/`  of Integer type that allows you to pass  `?number1=3543&number2=3478959`  or have a json object  `{ "number1": 3543, "number2": 3478959 }`  - especially the JSON example in a type shows that it's important to know that it's a number.

**Comment by timonrey** [comment]
We decided to vote for the first suggestion in @nkuehn comment above.
